lib/queue.js

```diff
@@ -642,32 +642,19 @@ Queue.prototype.startMoveUnlockedJobsToWait = function() {
   }
 };
 
-
-Queue.prototype.processJobs = function(resolve, reject, options){
+Queue.prototype.processJobs = function(resolve, reject){
   var _this = this;
-  var processJobs = this.processJobs.bind(this, resolve, reject, options);
+  var processJobs = this.processJobs.bind(this, resolve, reject);
   if(!this.closing){
-    var idleTime = 0;
-    var obeyRateLimit = function() {};
-
-    if (options.rateLimiter) {
-      obeyRateLimit = function() {
-        return options.rateLimiter()
-          .then(function(timeLeft) {
-            if (!timeLeft) { return; }
-            idleTime += timeLeft;
-            return Promise.delay(timeLeft)
-              .then(obeyRateLimit);
-          });
-      };
-    }
+    var startTime = new Date().getTime();
 
     process.nextTick(function(){
       (_this.paused || Promise.resolve())
-        .then(obeyRateLimit)
         .then(_this.getNextJob)
         .then(function(job) {
-          job.idleTime = idleTime;
+          if (job !== undefined) {
```

Comment:
@leahciMic ok, so what you are talking about is a mechanism that reports how long time a job is waiting in the queue, and if that time is too long it can spin new worker instances. That is a  pretty cool concept actually. Used together with AWS for example, the queue could just grow or shrink as needed to guarantee a given wait target. I think thought that this should be handled in a separate issue, and it will also allow us to better comment on different ways to accomplish it.